src/System.Diagnostics.Context/SpanContext.cs

```diff
@@ -4,18 +4,25 @@ namespace System.Diagnostics.Context
 {
     public class SpanContext
     {
-        public readonly string SpanId;
```

Comment:
In my review I am coming to doubt the value of SpanContext.   It seems like the state /properties can be rolled into Span.  

The only value of SpanContext was that it clearly differentiated between the state that flows to children and the state that does not.   However I think that that is just Baggage (and always will be), so the value of making a separate class is very low (also we need to make one of these per span, which is an pretty unfortunate inefficiency).  

Thus I think we should get rid of it.